cli/commands/catalog/tui/tui.go

```diff
@@ -8,10 +8,27 @@ import (
 	tea "github.com/charmbracelet/bubbletea"
 
 	"github.com/gruntwork-io/terragrunt/cli/commands/catalog/module"
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/internal/strict"
 	"github.com/gruntwork-io/terragrunt/options"
 )
 
 func Run(ctx context.Context, modules module.Modules, opts *options.TerragruntOptions) error {
+	if opts.ScaffoldRootFileName == "" {
```

Comment:
You also check this option in `cli/commands/catalog/command.go` `Action` flag func. Maybe it would be better to combine them and keep them in this func?